### DISCUSSION TITLE
Redbean lua: why isn't `sqlite3.backup_init` function available?

### DISCUSSION THREAD

**Discussion by bjornbm** [jart/cosmopolitan#1326] [Q&A] [answered]
Why isn't the function [`sqlite3.backup_init`](http://lua.sqlite.org/index.cgi/doc/tip/doc/lsqlite3.wiki#sqlite3_backup_init) available in the lsqlite3 library in redbean?

I guess I can extend the question to the other functions that seem to be missing (`sqlite.complete`, `sqlite.temp_directory`, `sqlite.open_ptr`), but `sqlite.backup_init` was the one I was looking for right now.

FYI, here are the functions I see:

```
$ redbean -i
>: require'lsqlite3'
{
  …
  __newindex="func@0x80001d278",
  config="func@0x80001d2c8",
  lversion="func@0x80001d238",
  open="func@0x80001f5d8",
  open_memory="func@0x80001f598",
  version="func@0x80001d538"
}
```

Thanks!

**Comment by pkulchenko** [comment, marked as answer]
It looks like it was removed in 88806b79 to keep the code more lean. Would [vacuum into](https://www.sqlite.org/lang_vacuum.html) work for your use case? This is what I'm using for DB backups.

**Reply by bjornbm** [reply]
[Vacuum into](https://www.sqlite.org/lang_vacuum.html) works perfectly! Thanks @pkulchenko for educating me about its existence. :)

Btw the commit you referred to 88806b7 seems to indicate that VACUUM was also disabled but I guess it was reinstated somewhere between then and #353. 🤷

**Reply by pkulchenko** [reply]
I did look into it as part of the discussion in #353 and I think that comment is wrong, as some of the functionality was removed, but not VACUUM or ALTER. You should be good with this and even older versions of redbean to use VACUUM INTO.

**Reply by bjornbm** [reply]
The commit itself also seemed to indicate disabling (`-DSQLITE_OMIT_VACUUM`) but I did not dig further and it's all the same to me as I am on latest redbean. :)
